aplib.net-demo/Assets/Scripts/MainMenu/MenuButtons.cs

```diff
@@ -1,48 +1,120 @@
 using UnityEngine;
 using UnityEngine.SceneManagement;
 
-public class MenuPlayButton : MonoBehaviour
+/// <summary>
+/// This class houses all the settings and menu UI buttons/methods.
+/// It is called CanvasManager as it manages all the UI canvi.
+/// </summary>
+public class CanvasManager : MonoBehaviour
 {
     /// <summary>
     /// Reference to the menu canvas.
     /// </summary>
     public GameObject menuCanvas;
 
     /// <summary>
-    /// Reference to the settings canvas.
+    /// Reference to the settings canvas of the menu.
     /// </summary>
-    public GameObject settingCanvas;
+    public GameObject settingMenuCanvas;
 
     /// <summary>
-    /// To ensure the settings and menu UI aren't on on the same time.
+    /// Reference to the settings canvas of the game.
     /// </summary>
-    public bool isOnSettings = false;
+    public GameObject settingGameCanvas;
 
+    /// <summary>
+    /// To ensure the menu settings and menu UI aren't on on the same time.
+    /// </summary>
+    public bool isOnMenuSettings = false;
+
+    /// <summary>
+    /// To ensure the game settings and menu UI aren't on on the same time.
+    /// </summary>
+    public bool isOnGameSettings = false;
+
+    /// <summary>
+    /// This bool communicates with the mouse lock script to enable and disable the cursor.
+    /// </summary>
+    public bool IsCursurNeeded => isOnGameSettings;
```

Comment:
Deze met je even met `f2` of `ctrl + r, ctrl + r` renamen maar dit moet Cursor zijn.

Reply:
Hoe??? D:

Reply:
idk it just happened